vms/evm/subscriber.go

```diff
@@ -154,25 +159,39 @@ func (s *subscriber) ProcessFromHeight(height *big.Int) error {
 		return err
 	}
 
-	// Cap the number of blocks to process to MaxBlocksToProcess
-	toBlock := big.NewInt(0).SetUint64(latestBlock)
-	if height.Cmp(big.NewInt(0).Sub(toBlock, big.NewInt(MaxBlocksToProcess))) < 0 {
-		s.logger.Warn(
-			fmt.Sprintf("Requested to process too many blocks. Processing only the most recent %d blocks", MaxBlocksToProcess),
-			zap.String("requestedBlockHeight", height.String()),
-			zap.String("latestBlockHeight", toBlock.String()),
-			zap.String("chainID", s.chainID.String()),
-		)
-		height = big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))
+	bigLatestBlock := big.NewInt(0).SetUint64(latestBlock)
+
+	// helper function to determine the upper bound of a block range given
+	// the lower bound:
+	toBlock := func(fromBlock *big.Int) (toBlock *big.Int) {
+		toBlock = big.NewInt(0).Add(fromBlock, big.NewInt(MaxBlocksPerRequest))
+		if toBlock.Cmp(bigLatestBlock) > 0 {
+			toBlock.Set(bigLatestBlock)
+		}
+		return
```

Comment:
If we're going to have this as a helper function, can we move it out to it's own function and call it `calculateToBlock`? That should hopefully make easier to read and easier to unit test.

That being said, I don't think a helper function is needed in this case. Could we make the for loop:
```
blockIncrement := big.NewInt(MaxBlocksPerRequest + 1)
for fromBlock := big.NewInt(0).Set(height); fromBlock.Cmp(bigLatestBlock) <= 0; fromBlock.Add(fromBlock, blockIncrement) {

Reply:
Doesn't this get us `MaxBlocksPerRequest + 1` blocks per request?

Reply:
@michaelkaplan13 thank you for pushing this further. I took your suggestion in https://github.com/ava-labs/awm-relayer/pull/89/commits/20d14f5d376b3acd141b36f0e649b89c99eb37af

@geoff-vball no because this is just with regards to iterating the `fromBlock` param from one request to the next, not the `toBlock` param

Reply:
I think @geoff-vball is correct here actually. @cam-schultz also called it out and suggested the fix in this comment https://github.com/ava-labs/awm-relayer/pull/89/files?diff=split&w=1#r1402690097

Reply:
thanks for connecting the dots here; conversation continued over there (https://github.com/ava-labs/awm-relayer/pull/89#discussion_r1408132909)